sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -500,13 +506,52 @@ def _add_argparse_args(cls, parser):
                         choices=['COST_OPTIMIZED', 'SPEED_OPTIMIZED'],
                         help='Set the Flexible Resource Scheduling mode')
 
+  def __getattr__(self, name):
+    if name in ["temp_location"]:
+      return self._get_temp_location()
+    else:
+      return super(GoogleCloudOptions, self).__getattr__(name)
+
+  def __setattr__(self, name, value):
+    if name in ["temp_location"]:
+      self._set_temp_location(value)
+    else:
+      super(GoogleCloudOptions, self).__setattr__(name, value)
+
+  @deprecated(
+      since='2.16.0',
+      custom_message=(
+          'GoogleCloudOptions.temp_location is deprecated since %since%. '
+          'Use GoogleCloudOptions.gcp_temp_location instead.'))
```

Comment:
Possibly also suggest StandardOptions.temp_location?

Reply:
I updated the message to read:

```
          'GoogleCloudOptions.temp_location is deprecated since %since%. '
          'Please use StandardOptions.temp_location or '
          'GoogleCloudOptions.gcp_temp_location, as appropriate.'))
```

Does that sound better?